src/utils.js

```diff
@@ -0,0 +1,95 @@
+/* Various Utility Functions */
+
+/* array: array to deep flatten, preserving order of elements
+ *     All elements in sub-arrays are placed in the top level
+ *     Arrays wrapped inside objects are not altered
+ *     TO DO, POSSIBLY: Place this and other functions (scrubObject, scrubArray from defaultOptions.js) in utils.js
+ */
+export function deepFlattenArray(passedArray) {
```

Comment:
I haven't dug into the details of what these functions are doing but at least one (deepFlattenArray) seems like lodash already provides this. https://lodash.com/docs/4.17.4#flattenDeep. I know you're just moving things around and adding tests, but could we replace this and others with lodash?